cmd/resource-quota-check/resources.go

```diff
@@ -0,0 +1,204 @@
+	reportErr := kh.ReportSuccess()
+	if reportErr != nil {
+		log.Fatalln("error reporting success to kuberhealthy:", reportErr.Error())
+	}
+}
+
+// examineResourceQuotas looks at the resource quotas and makes reports on namespaces that meet or pass the threshold.
+func examineResourceQuotas(namespaceList *v1.NamespaceList) chan []string {
+	resultChan := make(chan []string)
+
+	resourceQuotasJobChan := make(chan *Job, len(namespaceList.Items))
+	resourceQuotaErrorsChan := make(chan string, len(namespaceList.Items))
+
+	go fillJobChan(namespaceList, resourceQuotasJobChan)
+
+	go func(jobs chan *Job, results chan string) {
+
+		errors := make([]string, 0)
+		waitGroup := sync.WaitGroup{}
+
+		for job := range jobs {
+			waitGroup.Add(1)
+			log.Debugln("Starting worker for", job.namespace, "namespace.")
+			go createWorkerForNamespaceResourceQuotaCheck(job.namespace, results, &waitGroup)
+		}
+
+		go func(wg *sync.WaitGroup) {
+			log.Debugln("Waiting for workers to complete.")
+			wg.Wait()
+			log.Debugln("Workers done. Closing resource quota examination channel.")
+			close(results)
+		}(&waitGroup)
+
+		for err := range results {
+			errors = append(errors, err)
+		}
+		resultChan <- errors
+
+		return
+	}(resourceQuotasJobChan, resourceQuotaErrorsChan)
+
+	return resultChan
+}
+
+// createWorkerForNamespaceResourceQuotaCheck looks at the resource quotas for a given namespace and creates error messages
+// if usage is over a threshold.
+func createWorkerForNamespaceResourceQuotaCheck(namespace string, quotasChan chan string, wg *sync.WaitGroup) {
+	defer wg.Done()
+	defer log.Debugln("worker for", namespace, "namespace is done!")
+	switch {
+	// If whitelist and blacklist options are enabled, look at all whitelisted namespaces but the blacklisted namespaces.
+	case whitelistOn && blacklistOn:
+		// Skip non-whitelisted namespaces.
+		if !contains(namespace, whitelistNamespaces) {
+			log.Infoln("Skipping", namespace, "namespace.")
+			return
+		}
+		// Skip blacklisted namespaces.
+		if contains(namespace, blacklistNamespaces) {
+			log.Infoln("Skipping", namespace, "namespace.")
+			return
+		}
+		examineResouceQuotasForNamespace(namespace, quotasChan)
+		return
+	// If whitelist option is enabled, only look at the specified namespaces.
+	case whitelistOn:
+		// Skip non-whitelisted namespaces.
+		if !contains(namespace, whitelistNamespaces) {
+			log.Infoln("Skipping", namespace, "namespace.")
+			return
+		}
+		examineResouceQuotasForNamespace(namespace, quotasChan)
+		return
+	// If blacklist option is enabled, ignore the specified namespaces.
+	case blacklistOn:
+		// Skip blacklisted namespaces.
+		if contains(namespace, blacklistNamespaces) {
+			log.Infoln("Skipping", namespace, "namespace")
+			return
+		}
+		examineResouceQuotasForNamespace(namespace, quotasChan)
+		return
+	// By default, look at all namespaces.
+	default:
+		examineResouceQuotasForNamespace(namespace, quotasChan)
+		return
+	}
```

Comment:
I bet you can do this much simpler.  Try using a small number of `if` statements and using `return` where appropriate to skip the namespace.

Try getting rid of your `whitelistOn` and `blacklistOn` globals entirely if you're up for it.

If you don't feel like the refactor, I think this approach is fine for a first release.